### PR TITLE
Qualcomm AI Engine Direct - add cli tool for QNN artifacts

### PR DESCRIPTION
**Pull request by haowhsu-quic**
Summary:
- cli tool for deploying precompiled model library / context bin onto executorch runtime
- refactor & mionr fixes

### CODE REVIEW

**Comment by haowhsu-quic** [comment]
Hi @cccclai, this PR is a CLI tool for helping convert pre-generated QNN artifacts (`*.so` / `*.bin`) into `.pte` program. We think this will make user engage with Qualcomm AIHUB more smoothly.
Please have a look, thank you!

File: examples/qualcomm/qaihub_scripts/utils/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# CLI Tool for Compile / Deploy Pre-Built QNN Artifacts
+
+An easy-to-use tool for generating / executing .pte program from pre-built model libraries / context binaries from Qualcomm AI Engine Direct. Tool is verified with [host environement](../../../../docs/source/build-run-qualcomm-ai-engine-direct-backend.md#host-os).
```
Comment (cccclai) [inline examples/qualcomm/qaihub_scripts/utils/README.md:3]: Is it generic for all models from ai hub?

**Review by cccclai — APPROVED** [review]
lgtm. Just one question related to the ai hub models

File: examples/qualcomm/qaihub_scripts/utils/README.md (hunk 1 above)
Comment (haowhsu-quic) [reply examples/qualcomm/qaihub_scripts/utils/README.md:3]: Yes, artifacts from AIHUB related to QNN are delivered with `.so` format. Only large generative AI models are shipped with context binaries.
Both of them could be transformed into `.pte` program with this tool.

**Pull request merged**